### DISCUSSION TITLE
Use domains in "disallowed_clients" and "allowed_clients" instead of IP address

### DISCUSSION THREAD

**Discussion by Issam2204** [AdguardTeam/AdGuardHome#5699] [Q&A]
I'm hosting AdGuard Home on a VPS on the cloud (all secured with encryption and TLS). However, I see that sometimes new clients shows-up (normally only making 1 request) and I'd like to avoid that.

 I'd like to only allow my home router to connect to AdGuard: I'm wondering if the "disallowed_clients" and "allowed_clients" settings can be setup using domain names (especially the "allowed_clients" option). Why? Because my ISP IP address can change, but via DDNS I can always refer to my home network.

Thanks for any suggestion or help!

**Comment by ainar-g** [comment]
If your router supports DoH, DoQ, or DoT, a better way to ensure that only the router can access it would be to use the [ClientIDs](https://github.com/AdguardTeam/AdGuardHome/wiki/Clients#clientid).  Blocking by a client's domain name would be less precise and less fast, as that would require making rDNS queries and awaiting their results.

**Comment by nealhead** [comment]
I would like this is well. I am hosting an Adguard on a public instance and would like to whitelist domains for those environments that use dynamic IPs.
